app/models/spree/gateway/amazon.rb

```diff
@@ -145,6 +145,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+
+      response = @mws.refund(response_code, order.number, order.total, order.currency)
```

Comment:
Could we possible use `"#{order.number}-cancel` for the second parameter?  Just as a nicety to make things more understandable when looking in the Amazon admin & etc.